server/memphis_handlers_consumers.go

```diff
@@ -233,22 +233,73 @@ func (s *Server) createConsumerDirect(c *client, reply string, msg []byte) {
 		return
 	}
 
-	newConsumer := models.Consumer{
-		ID:               primitive.NewObjectID(),
-		Name:             name,
-		StationId:        station.ID,
-		Type:             consumerType,
-		ConnectionId:     connectionIdObj,
-		CreatedByUser:    connection.CreatedByUser,
-		ConsumersGroup:   consumerGroup,
-		IsActive:         true,
-		CreationDate:     time.Now(),
-		IsDeleted:        false,
-		MaxAckTimeMs:     int64(ccr.MaxAckTimeMillis),
-		MaxMsgDeliveries: ccr.MaxMsgDeliveries,
+	if ccr.StartConsumeFromSequence > 1 && ccr.LastMessages > -1 {
+		errMsg := "Consumer creation options can't contain both startConsumeFromSequence and lastMessages"
+		serv.Errorf("createConsumerDirect: " + errMsg)
+		respondWithErr(s, reply, err)
+		return
+	}
+
+	var newConsumer models.Consumer
+	if ccr.StartConsumeFromSequence > 1 {
```

Comment:
in each block you are re-write the entire parameters you can inject the relevant where needed